### PR TITLE
feat: VPA for VMAlert

### PR DESCRIPTION
**Pull request by rakesh-von**
Reason for this change - For components like VMAlert where scaling horizontally doesn't solve issues with processing alerts, scaling vertically is the only option to deal with scaling problems. 

Tested locally by installing VMAlert with and without VPA enabled from values file and both the times the installation worked.

### CODE REVIEW

**Comment by rakesh-von** [comment]
HI @zekker6 - can i get a review of this please?

File: charts/victoria-metrics-alert/templates/vpa.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{{- if .Values.server.VerticalPodAutoscaler.enabled }}
```
Comment (Haleygo) [inline charts/victoria-metrics-alert/templates/vpa.yaml:1]:
```suggestion
{{- if and (.Capabilities.APIVersions.Has "autoscaling.k8s.io/v1") (.Values.server.verticalPodAutoscaler.enabled) }}```

File: charts/victoria-metrics-alert/values.yaml [hunk 2]
```diff
@@ -272,6 +272,26 @@ server:
     alerts:
       groups: []
 
+  # -- Vertical Pod Autoscaler
+  VerticalPodAutoscaler:
```
Comment (Haleygo) [inline charts/victoria-metrics-alert/values.yaml:276]:
```suggestion
  verticalPodAutoscaler:
```

**Review by Haleygo — COMMENTED** [review]
Could you also help adding the feature to [changelog](https://github.com/VictoriaMetrics/helm-charts/blob/fix-multitenancy-mode/charts/victoria-metrics-alert/CHANGELOG.md)?

**Comment by rakesh-von** [comment]
@Haleygo - made changes as suggested.

how do i know if VPA is supported in `autoscaling/v2` api group? can you help me find the proper link?

**Comment by Haleygo** [comment]
> @Haleygo - made changes as suggested.
> 
> how do i know if VPA is supported in `autoscaling/v2` api group? can you help me find the proper link?

It's the same as v1 check, like
```
(.Capabilities.APIVersions.Has "autoscaling/v2")
```
And as I can see, [autoscaling/v2](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) is stable since k8s v1.23, so the most of the users would use `autoscaling/v2` instead of `autoscaling/v1`.

**Comment by rakesh-von** [comment]
> > @Haleygo - made changes as suggested.
> > how do i know if VPA is supported in `autoscaling/v2` api group? can you help me find the proper link?
> 
> It's the same as v1 check, like
> 
> ```
> (.Capabilities.APIVersions.Has "autoscaling/v2")
> ```
> 
> And as I can see, [autoscaling/v2](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) is stable since k8s v1.23, so the most of the users would use `autoscaling/v2` instead of `autoscaling/v1`.

I checked the k8s autoscaler code and do not find VPA using `autoscaling/v2` https://github.com/search?q=repo%3Akubernetes%2Fautoscaler++path%3Avertical-pod-autoscaler%2Fpkg+autoscaling%2Fv2&type=code instead uses `autoscaling.k8s.io/v1` https://github.com/search?q=repo%3Akubernetes%2Fautoscaler++path%3Avertical-pod-autoscaler%2Fpkg+autoscaling.k8s.io%2Fv1&type=code

I tested in my local cluster as well with EKS version `v1.28.9-eks-036c24b` and VPA CRD deployed and running fine with `autoscaling.k8s.io/v1` VPA objects. I get below error when i use `autoscaling/v2`
```
helm install my-vpa-test .
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: resource mapping not found for name: "my-vpa-test-victoria-metrics-alert-server" namespace: "default" from "": no matches for kind "VerticalPodAutoscaler" in version "autoscaling/v2"
ensure CRDs are installed first
```

Please suggest.

**Comment by rakesh-von** [comment]
@Haleygo - can i have your comments/review please? we are waiting for this change.

**Comment by Haleygo** [comment]
> > > @Haleygo - made changes as suggested.
> > > how do i know if VPA is supported in `autoscaling/v2` api group? can you help me find the proper link?
> > 
> > 
> > It's the same as v1 check, like
> > ```
> > (.Capabilities.APIVersions.Has "autoscaling/v2")
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And as I can see, [autoscaling/v2](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) is stable since k8s v1.23, so the most of the users would use `autoscaling/v2` instead of `autoscaling/v1`.
> 
> I checked the k8s autoscaler code and do not find VPA using `autoscaling/v2` https://github.com/search?q=repo%3Akubernetes%2Fautoscaler++path%3Avertical-pod-autoscaler%2Fpkg+autoscaling%2Fv2&type=code
> 
> instead uses `autoscaling.k8s.io/v1` https://github.com/search?q=repo%3Akubernetes%2Fautoscaler++path%3Avertical-pod-autoscaler%2Fpkg+autoscaling.k8s.io%2Fv1&type=code
> 
> I tested in my local cluster as well with EKS version `v1.28.9-eks-036c24b` and VPA CRD deployed and running fine with `autoscaling.k8s.io/v1` VPA objects. I get below error when i use `autoscaling/v2`
> 
> ```
> helm install my-vpa-test .
> Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: resource mapping not found for name: "my-vpa-test-victoria-metrics-alert-server" namespace: "default" from "": no matches for kind "VerticalPodAutoscaler" in version "autoscaling/v2"
> ensure CRDs are installed first
> ```
> 
> Please suggest.

Sorry, I confused it with hpa before.

**Review by Haleygo — APPROVED** [review]
LGTM

**Comment by Haleygo** [comment]
@rakesh-von JFYI, [victoria-metrics-alert-0.9.11](https://github.com/VictoriaMetrics/helm-charts/releases/tag/victoria-metrics-alert-0.9.11) has already been released with this feature.

**Pull request merged**